docs/en/advanced_guides/datasets.md

```diff
@@ -0,0 +1,147 @@
+# Dataset
+
+In this document, we will introduce the design of each datasets in MMGeneration and how users to add their own dataset.
+
+In 1.x version of MMGeneration, all datasets are inherited from BaseDataset.
+Each dataset load the data list by `load_data_list` and call `prepare_data` in `__getitem__` to load the data info. In `prepare_data`, `get_data_info` and transforms pipeline will be called progressively.
```

Comment:
Specifically, a data processing pipeline in a dataset consists of the following steps: 
1. loading the list of dataset samples, implemented in `prepare_data`
2. loading dataset information, implemented in `__getitem__`
3. xxxxx